plugins/dell-k2/README.md

```diff
@@ -96,6 +96,16 @@ the dock's Type-C cable from the host, at which point they will take effect.
 
 The vendor ID is set from the USB vendor, in this instance set to `USB:0x413C`
 
+## Quirk Use
+
+This plugin uses the following plugin-specific quirks:
+
+### `uod-off`
```

Comment:
I think you're mixing up private flags and quirks.  Which one do you actually want?  If it's something you would add to the quirk file, you'll need to have a matching callback for it (look for example how `DellDockUnlockTarget` works).

Reply:
Ooh, I wanted to use [device flag](https://lvfs.readthedocs.io/en/latest/metainfo.html#device-flags) so it can be added in the firmware metainfo.xml file.

Reply:
Yeah so use that flag for usable while updating to control everything. Then you don't need a new flag anywhere to decide this.

Reply:
Adding flags from component MD at present accept private flags only, as currently is: 
```
fu_device_add_private_flag(FU_DEVICE(self), FU_DEVICE_PRIVATE_FLAG_MD_SET_FLAGS);
```

The flag `usable-during-update` is a non-private device flag.  Looked through the existing private flags I can't find one that is closer to this need.

Reply:
Yeah I guess it makes sense that it can only set private flags.  But you can add a private flag and then if that flag is set from metadata then add `usable-during-update` from the plugin.

Reply:
Thinking twice, the device in fact will be usable during the update process regardless with/without the `uod-off` flag, it's just whether the command should be sent or no.  In case no it requires a full power cycle to the dock to trigger the update.

If we want the `usable-during-update` do what it should do, perhaps we shouldn't toggle it.

Reply:
`uod-off` is sort of a temporary fix to address the update issue, when the ec feature is implemented than this flag should be removed from the MD.